### ISSUE TITLE
Question regarding dnscrypt-blacklist-domains.txt syntax

### ISSUE THREAD

**Issue by zymase** [jedisct1/dnscrypt-proxy#545]
I've searched, I promise! but haven't found the answer to the syntax concerning the user's domains' blacklist, i.e dnscrypt-blacklist-domains.txt. I'm aware this is an issue thread but it's the only place in town I know to get information regarding dnscrypt-proxy ...

What I wish to know is if adding to this list, i.e domain.com includes or not *.domain.com or do I have to set both:
domain.com
*.domain.com

Generally speaking, can I use * (also ? (question mark) for one character only), i.e. is the following correct:
[star] . 30[star]br . [star] (I cannot write it such as here...)

which would include among so many others,
ad.doubleclick.net.14303.9065.302br.net
304br.net
etc etc

Thanks again!

**Comment by jedisct1** [comment]
This is documented: https://github.com/jedisct1/dnscrypt-proxy/wiki/Filtering#names-rules-set

But maybe not visible enough. Having some examples in the configuration file itself may not be a bad idea.

So, `*.example.com` includes `example.com`.

However, stars can only be at the beginning and/or at the end of the pattern.
So, `*.30*br.*` doesn't work, as this kind of pattern could be really slow to match.

However, don't forget that you can also block by IP address. For example all these `30*br.net` names seem to resolve to `69.172.216.55` or `69.172.216.56`. So, blocking the IPs is likely to solve the problem and be way more efficient than tracking their domains :)

**Comment by zymase** [comment]
Thanks, again, @jedisct1 
Mea culpa, I hadn't searched well enough to find what is explicitly expressed indeed in the 'Names rules set' page you mention. I guess as often I lacked zeal in my quest, would be a surprise if I change at my age!

OK, I got it. I also understand that "stars can only be at the beginning and/or at the end of the pattern.". I'm on my way to correct my domains blacklist.

I'm closing the thread because I see that as a possibility when I would have thought you only could close your pages. If it' OK with you then it is with me : a closed case!
